Sources/Commands/SwiftTestTool.swift

```diff
@@ -1320,6 +1320,7 @@ private extension Basics.Diagnostic {
 ///
 /// - Returns: The paths to the build test products.
 private func buildTestsIfNeeded(swiftTool: SwiftTool, buildParameters: BuildParameters, testProduct: String?) throws -> [BuiltTestProduct] {
+    // FIXME: Don't create the build system multiple times, that's extremely wasteful.
```

Comment:
This is a real problem, but I am not going to address it as part of this PR. Initially, this was a problem for me during development, but it has since been solved.